lib/sharezone_localizations/lib/src/app_locales.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+import 'package:sharezone_localizations/sharezone_localizations.dart';
+
+/// A wrapper around [Locale] to provide differentiation between the system
+/// locale and the user selected locale.
+enum AppLocales {
+  system,
+  en,
+  de;
+
+  /// Converts the enum value to a [Locale] object.
+  Locale toLocale() {
+    return switch (this) {
+      system => getSystemLocale(),
+      en => SharezoneAppLocales.en,
+      de => SharezoneAppLocales.de,
+    };
+  }
+
+  /// Returns the name of the locale in the native language, e.g. "Deutsch" for
+  /// the [AppLocales.de] enum value.
+  String getNativeName(BuildContext context) {
+    return switch (this) {
+      system => context.sl.language_system_name,
+      en => 'English',
+      de => 'Deutsch',
+    };
+  }
+
+  /// Returns the name of the locale in the currently selected language, e.g.
+  /// "German" for the [AppLocales.de] enum value when the app is in English.
+  String getTranslatedName(BuildContext context) {
+    return switch (this) {
+      system => context.sl.language_system_name,
+      en => context.sl.language_en_name,
+      de => context.sl.language_de_name,
+    };
+  }
+
+  /// The system-reported default locale of the device.
+  static Locale getSystemLocale() {
+    return PlatformDispatcher.instance.locale;
+  }
+
+  static AppLocales fromMap(Map<String, dynamic>? map) {
+    if (map == null || map['isSystem'] as bool) {
+      return system;
+    }
+    return fromLanguageTag(map['languageTag']);
+  }
+
+  Map<String, dynamic> toMap() {
+    return {
+      'isSystem': isSystem(),
+      'languageTag': _toLanguageTag(),
+    };
```

Comment:
This is a proposal on how we could store it in the database. We need the `isSystem` to differentiate if the language is the system language or the selected language.